src/web/nextui/src/app/eval/ResultsCharts.tsx

```diff
@@ -415,13 +426,269 @@ function MetricChart({ table }: ChartProps) {
   return <canvas ref={metricCanvasRef} style={{ maxHeight: '300px' }}></canvas>;
 }
 
-function ResultsCharts({ columnVisibility }: ResultsChartsProps) {
+function PerformanceOverTimeChart({ table, evalId, config, datasetId }: ChartProps) {
+  const lineCanvasRef = useRef(null);
+  const lineChartInstance = useRef<Chart | null>(null);
+  const [dataset, setDataset] = useState<TestCasesWithMetadata>();
+
+  interface PointMetadata {
+    evalId: string;
+    highlight: boolean;
+    label: string;
+    date: string;
+    score: string;
+  }
+
+  interface Point {
+    x: number;
+    y: number;
+    metadata: PointMetadata;
+  }
+
+  const fetchDataset = async (id: string) => {
```

Comment:
let's move this and `ordinalSuffixOf` out of the component function

Reply:
✅